modules/cockroachdb/cockroachdb.go

```diff
@@ -252,29 +254,57 @@ func connString(opts options, host string, port nat.Port) string {
 		Scheme:   "postgres",
 		User:     user,
 		Host:     net.JoinHostPort(host, port.Port()),
-		Path:     opts.Database,
+		Path:     c.database,
 		RawQuery: params.Encode(),
 	}
 
-	return u.String()
+	cfg, err := pgx.ParseConfig(u.String())
+	if err != nil {
+		return nil, fmt.Errorf("parse config: %w", err)
+	}
+
+	cfg.TLSConfig = tlsConfig
+
+	return cfg, nil
 }
 
-func connTLS(opts options) (*tls.Config, error) {
-	if opts.TLS == nil {
-		return nil, ErrTLSNotEnabled
+// configure sets the CockroachDBContainer options from the given request and updates the request
+// wait strategies to match the options.
+func (c *CockroachDBContainer) configure(req *testcontainers.GenericContainerRequest) error {
+	c.database = req.Env[envDatabase]
+	c.user = req.Env[envUser]
+	c.password = req.Env[envPassword]
+
+	var insecure bool
+	for _, arg := range req.Cmd {
+		if arg == insecureFlag {
+			insecure = true
+			break
+		}
 	}
 
-	keyPair, err := tls.X509KeyPair(opts.TLS.ClientCert, opts.TLS.ClientKey)
-	if err != nil {
-		return nil, err
-	}
+	// Walk the wait strategies to find the TLS strategy and either remove it or
+	// update the client certificate files to match the user and configure the
+	// container to use the TLS strategy.
+	if err := wait.Walk(&req.WaitingFor, func(strategy wait.Strategy) error {
+		if cert, ok := strategy.(*wait.TLSStrategy); ok {
+			if insecure {
+				// If insecure mode is enabled, the certificate strategy is removed.
+				return errors.Join(wait.VisitRemove, wait.VisitStop)
```

Comment:
I see now it pretty well, and I like it. Thanks!

follow-up: we probably need an example like this one in https://golang.testcontainers.org/features/wait/walk/

Reply:
Yep planning to do a follow up PR for that.